plugin-core/src/main/java/appland/actions/PluginStatus.java

```diff
@@ -0,0 +1,114 @@
+package appland.actions;
+
+import appland.cli.AppLandDownloadService;
+import appland.cli.CliTool;
+import appland.javaAgent.JavaAgentStatus;
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.actionSystem.UpdateInBackground;
+import com.intellij.openapi.application.WriteAction;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.fileEditor.FileEditorManager;
+import com.intellij.openapi.progress.PerformInBackgroundOption;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.DumbAware;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.project.ProjectUtil;
+import com.intellij.openapi.vfs.VfsUtil;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.util.concurrency.annotations.RequiresBackgroundThread;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.IOException;
+import java.util.Objects;
+
+
+public class PluginStatus extends AnAction implements UpdateInBackground, DumbAware {
+    private static final Logger LOG = Logger.getInstance(PluginStatus.class);
+    private static final String STATUS_REPORT_FILENAME = "appmap_status_report.md";
+
+    @Override
+    public void actionPerformed(@NotNull AnActionEvent event) {
+        createStatusReport(Objects.requireNonNull(event.getProject()));
+    }
+
+    @RequiresBackgroundThread
+    public static void createStatusReport(@NotNull Project project) {
+        VirtualFile projectDir = ProjectUtil.guessProjectDir(project);
+        if (projectDir == null) return;
+
+        new Task.Backgroundable(project, "Generating AppMap Plugin status report...",  true, PerformInBackgroundOption.ALWAYS_BACKGROUND) {
+
+            private volatile VirtualFile statusReportFile;
+
+            @Override
+            public void run(@NotNull ProgressIndicator indicator) {
+                String reportText = statusReportText(indicator);
+
+                statusReportFile = WriteAction.computeAndWait(() -> {
+                    try {
+                        var file = projectDir.createChildData(this, STATUS_REPORT_FILENAME);
+                        VfsUtil.saveText(file, reportText);
+                        return file;
+                    } catch (IOException e) {
+                        LOG.error(e);
+                        return null;
+                    }
+                });
+            }
+
+            @Override
+            public void onSuccess() {
+                if (statusReportFile != null) {
+                    var editorManager = FileEditorManager.getInstance(project);
+                    editorManager.openFile(statusReportFile, true);
+                }
+            }
+        }.queue();
```

Comment:
I don't think we should create a new file on disk.
It's possible to create an in-memory file, which is not stored on disk and dropped when the editor is closed.
See below :arrow_down: how this could be done.

Please move ` "Generating AppMap Plugin status report..."` into the message bundle `appland.properties` and use something like this to show it: `AppMapBundle.get("pluginStatus.progressTitle")`.

```suggestion
        new Task.Backgroundable(project, "Generating AppMap Plugin status report...", true, PerformInBackgroundOption.ALWAYS_BACKGROUND) {
            private volatile VirtualFile statusReportFile;

            @Override
            public void run(@NotNull ProgressIndicator indicator) {
                statusReportFile = new LightVirtualFile(STATUS_REPORT_FILENAME, statusReportText(indicator));
            }

            @Override
            public void onSuccess() {
                if (statusReportFile != null) {
                    var editorManager = FileEditorManager.getInstance(project);
                    editorManager.openFile(statusReportFile, true);
                }
            }
        }.queue();
```

Reply:
Thanks!